### PR TITLE
feat(GDB-10848) Implement Cluster Management View

### PR DESCRIPTION
**Pull request by teodossidossev**
## WHAT:
Implemented a new cluster management modal view where all actions related to creating, deleting, and replacing cluster nodes will take place

## WHY:
Requirement

## HOW:
- Refactored URL validation logic in url-utils.js
- Enhanced the validate-url directive to handle protocol exclusions and empty value validations
- Added support for updating cluster groups with a modal dialog in the cluster management UI.
- Improved error handling and state management in cluster-context.service.js and related components.
- Updated cluster-management.controller.js to handle user actions and ensure proper state transitions.
- Applied CSS improvements for the cluster management UI, including styles for autocomplete and error messages.

### CODE REVIEW

File: src/js/angular/clustermanagement/templates/cluster-list.html [hunk 1]
```diff
@@ -0,0 +1,124 @@
+<div onto-loader-new ng-if="loader" class="create-cluster-loader" message="getLoaderMessage" size="100  "></div>
+
+<div class="table-responsive" ng-class="isEditMode() ? 'edit-mode': 'preview-mode'">
+    <div class="pull-right mb-1">
+        <button ng-click="addNode()" ng-disabled="editedNodeIndex !== undefined"
+                class="btn btn-primary add-node-btn"
+                gdb-tooltip="{{'cluster_management.update_cluster_group_dialog.actions.add_node_tooltip' | translate}}">
+            <em class="icon-plus"></em>{{'cluster_management.update_cluster_group_dialog.actions.add_node' | translate}}
```
Comment (svilenvelikov) [inline src/js/angular/clustermanagement/templates/cluster-list.html:8]: You can use the font awesome icons already. Donka prefers the `regular` once.

File: src/js/angular/clustermanagement/directives/cluster-list.directive.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+                }
+            };
+
+            // =========================
+            // Private functions
+            // =========================
+            const onClusterViewChanged = (cluster) => {
+                $scope.cluster = cluster;
+                $scope.clusterView = cluster.getAttached();
+                $scope.allSuggestions = cluster.getAvailable().map((node) => node.endpoint);
```
Comment (svilenvelikov) [inline src/js/angular/clustermanagement/directives/cluster-list.directive.js:153]: You can move/hide/ this and some other similar operations in the model itself. For example in this case you can have `cluster.getAvailableNodeEndpoints()`

File: src/js/angular/clustermanagement/templates/cluster-list.html [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            </tr>
+            </thead>
+            <tbody>
+            <tr ng-if="!clusterView.length">
+                <td colspan="5" class="no-data">
+                    {{'cluster_management.update_cluster_group_dialog.messages.no_cluster' | translate}}
+                </td>
+            </tr>
+
+            <tr ng-repeat-start="node in clusterView track by $index" ng-if="0"></tr>
```
[the remark this answers is not in the document]
Comment (teodossidossev) [reply src/js/angular/clustermanagement/templates/cluster-list.html:35]: for tags used for ng-repeat-start and ng-repeat-end I set ng-if="0", to let not be inserted in the page

[the remark this answers is not in the document]
Comment (svilenvelikov) [reply src/js/angular/clustermanagement/templates/cluster-list.html:35]: yeah, I figured it out, but forgot to write back. I've used it in some other views too

File: src/js/angular/clustermanagement/directives/cluster-list.directive.js (hunk 2 above)
Comment (teodossidossev) [reply src/js/angular/clustermanagement/directives/cluster-list.directive.js]: done

File: src/i18n/locale-en.json [hunk 4]
```diff
@@ -147,6 +148,30 @@
         "remove_nodes_dialog": {
             "title": "Remove nodes from cluster",
             "content": "Select nodes to be removed from existing cluster"
+        },
+        "update_cluster_group_dialog": {
```
[the remark this answers is not in the document]
Comment (teodossidossev) [reply src/i18n/locale-en.json:152]: sure

File: src/js/angular/clustermanagement/templates/cluster-list.html [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                    <div ng-if="!node.address">
+                        {{'cluster_management.update_cluster_group_dialog.new_node' | translate}}
+                    </div>
+                </td>
+                <td class="empty-cell"></td>
+                <td class="actions-cell">
+                    <div ng-if="editedNodeIndex === undefined" class="actions-group">
+                        <button ng-click="replaceNode($index)" class="btn btn-link replace-node-btn"
+                                gdb-tooltip="{{'cluster_management.update_cluster_group_dialog.actions.replace_node' | translate}}">
+                            <em class="icon-exchange"></em>
```
[the remark this answers is not in the document]
Comment (teodossidossev) [reply src/js/angular/clustermanagement/templates/cluster-list.html]: done

File: src/js/angular/clustermanagement/templates/cluster-list.html (hunk 1 above)
Comment (teodossidossev) [reply src/js/angular/clustermanagement/templates/cluster-list.html]: done

File: src/js/angular/clustermanagement/directives/cluster-list.directive.js [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+             * Filters suggestions based on user input for node's endpoint.
+             * @param {Object} node - The node object with an endpoint property.
+             */
+            $scope.filterSuggestions = (node) => {
+                const query = node.endpoint && node.endpoint.toLowerCase();
+                if (query) {
+                    $scope.suggestions = $scope.allSuggestions.filter((item) => item.toLowerCase().includes(query));
+                    $scope.showDropdown = true;
+                } else {
+                    $scope.showDropdown = false;
```
Comment (boyan-tonchev) [inline src/js/angular/clustermanagement/directives/cluster-list.directive.js:49]: If the endpoint is available, we calculate `$scope.suggestions.` However, if it is not, we might need to reset `$scope.suggestions` to its default state.

File: src/i18n/locale-en.json [hunk 7]
```diff
@@ -147,6 +148,30 @@
...
+            "title": "Update cluster group",
```
Comment (boyan-tonchev) [inline src/i18n/locale-en.json:153]: You also need to update test-cypress/fixtures/locale-en.json.

**Review by svilenvelikov — APPROVED** [review]
LGTM, but check @boyan-tonchev 's notes as well

File: src/i18n/locale-en.json (hunk 7 above)
Comment (teodossidossev) [reply src/i18n/locale-en.json:153]: done

File: src/js/angular/clustermanagement/directives/cluster-list.directive.js (hunk 6 above)
Comment (teodossidossev) [reply src/js/angular/clustermanagement/directives/cluster-list.directive.js:50]: fixed

File: src/js/angular/models/clustermanagement/cluster.js [hunk 8]
```diff
@@ -0,0 +1,347 @@
...
+
+/**
+ * Utility class for transforming cluster data to a view model format.
+ */
+class ClusterUtil {
+    /**
+     * Converts location and node data into a view model format.
+     * @param {Location} location - The location data.
+     * @param {Node} [node] - The node data (optional).
+     * @return {Object} The view model object.
```
[the remark this answers is not in the document]
Comment (teodossidossev) [reply src/js/angular/models/clustermanagement/cluster.js]: done

**Review by boyan-tonchev — APPROVED** [review]

File: src/js/angular/models/clustermanagement/cluster.js [hunk 9]
```diff
@@ -0,0 +1,347 @@
...
+        const links = json.links.map(Link.fromJSON);
+
+        return new ClusterModel(locations, json.hasCluster, nodes, links);
+    }
+}
+
+/**
+ * ViewModel to manage and present the cluster model data.
+ */
+export class ClusterViewModel {
```
[the remark this answers is not in the document]
Comment (teodossidossev) [reply src/js/angular/models/clustermanagement/cluster.js:231]: good catch! 
created appropriate view model and applied the change

**Pull request merged**